test/integration/controller/jobset_controller_test.go

```diff
@@ -516,6 +527,27 @@ func matchJobsSuspendState(js *jobset.JobSet, suspend bool) (bool, error) {
 	return true, nil
 }
 
+func matchJobsNodeSelectors(js *jobset.JobSet, nodeSelectors map[string]map[string]string) (bool, error) {
+	var jobList batchv1.JobList
+	if err := k8sClient.List(ctx, &jobList, client.InNamespace(js.Namespace)); err != nil {
+		return false, err
+	}
+	jobsChanged := 0
+	// Check if all jobs with name prefix js.Name+replicatedJobName have updated node selectors.
+	for _, job := range jobList.Items {
+		for index, nodeSelector := range nodeSelectors {
+			if strings.HasPrefix(job.Name, index) {
+				if reflect.DeepEqual(job.Spec.Template.Spec.NodeSelector, nodeSelector) {
+					jobsChanged++
+					continue
+				}
+				return false, nil
+			}
+		}
+	}
+	return len(nodeSelectors) <= jobsChanged, nil
```

Comment:
probably better to add up replica counts of each replicatedJob to calculate the number of jobs

Reply:
yes, but again you are assuming 1 job per replicatedJob, better to add up the replicatedJob.replicas of the replicatedJobs that we are updating. The name of those replicatedJobs exist as the `nodeSelectors` key:

```
wantJobsChanged := 0
for rj := js.ReplicatedJobs {
  if _, exists := nodeSelectors[rj.name]; exists {
    wantJobsChaned += rj.Replicas
  }
}
return wantJobsChanged == jobsChanged
```